src/library/scala/util/control/ControlThrowable.scala

```diff
@@ -33,3 +33,33 @@ package util.control
  *  @author Miles Sabin
  */
 trait ControlThrowable extends Throwable with NoStackTrace
+
+object ControlThrowable {
+  sealed trait CachedBase extends Throwable with ControlThrowable {
+    override def fillInStackTrace(): Throwable = super[Throwable].fillInStackTrace()
```

Comment:
so... this overrides the `fillInStackTrace` in `NoStackTrace`, so that calling `fillInStackTrace` on one of these does, but the subclasses all call the `Throwable` constructor that disables calling `fillInStackTrace`, so it's all OK? I'm confused by how this helps anything.

Reply:
The thought was that it will call the default `fillInStackTrace()`, but since we pass the constructor parameter, that method doesn't do anything. Theoretically this path would be better optimised by the JVM, since it's more common? idk.

However, `super[Throwable]` isn't valid anyway, so I'm probably going to override it to return `this`.